src/Microsoft.AspNet.Mvc.Core/SessionStateTempDataProvider.cs

```diff
@@ -61,11 +61,16 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 return null;
             }
 
-            var tempDataDictionary = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
             var session = context.Session;
+            if (session == null)
```

Comment:
There is already a check for that here. https://github.com/aspnet/Mvc/pull/2647/files#diff-a87ef667e5f25cf11336f7ebc2be8530L58